### PR TITLE
Deprecate `Validate::isAnything`

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This function just returns true and does nothing else.
| Type?             | improvement
| Category?         | IN
| BC breaks?        | no
| Deprecations?     | yes
| How to test?      | This is just deprecating a function, no tests needed.

### CODE REVIEW

File: classes/Validate.php [hunk 1]
```diff
@@ -61,8 +61,16 @@ public static function isIp2Long($ip)
         return preg_match('#^-?[0-9]+$#', (string) $ip);
     }
 
+    /**
+     * @deprecated since PrestaShop 8.1 and will be removed in Prestashop 9.0
+     */
     public static function isAnything()
     {
+        @trigger_error(
+            'This function is deprecated PrestaShop 8.1 and will be removed in Prestashop 9.0.',
```
Comment (nicosomb) [inline classes/Validate.php:70]: one word missing: `is deprecated in PrestaShop 8.1`.

Comment (FabienPapet) [reply classes/Validate.php:70]: done

**Review by nicosomb — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Review by lartist — DISMISSED** [review]

**Comment by eternoendless** [comment]
Consider removing the constraints [here](https://github.com/PrestaShop/PrestaShop/blob/8.0.0/classes/order/OrderPayment.php#L50-L59) and marking the test for deletion [here](https://github.com/PrestaShop/PrestaShop/blob/8.0.0/tests/Unit/Classes/ValidateCoreTest.php#L42-L45)

**Comment by Hlavtox** [comment]
Guys, if you are going to do something with those fields in OrderPayment, please check my suggestion here - https://github.com/PrestaShop/PrestaShop/issues/22120.

I think it would be good that instead of those fields `card_number`, `card_brand`, `card_expiration`, `card_holder`, there could be just a `details `field where could payment modules store any detailed JSON information about the payments. There are now million ways to pay on top of classic card payment, and modules don't use the `ps_order_payment` at all, because they can't store the things what they need there. Example of such information:

payment

File: classes/Validate.php [hunk 2]
```diff
@@ -61,8 +61,16 @@ public static function isIp2Long($ip)
         return preg_match('#^-?[0-9]+$#', (string) $ip);
     }
 
+    /**
+     * @deprecated since PrestaShop 8.1 and will be removed in Prestashop 9.0
```
Comment (jolelievre) [inline classes/Validate.php:65]: Ok to remove this one, but in this case you should remove its usage as well. Basically not setting a validation is equivalent as using this one.

The only use cases I could find are in `classes/order/OrderPayment.php` where you should remove all the `'validate' => 'isAnything',`

**Comment by FabienPapet** [comment]
@jolelievre @nicosomb @lartist Need your approval again :)

**Review by jolelievre — APPROVED** [review]
Thanks @FabienPapet

**Review by nicosomb — APPROVED** [review]

**Review by lartist — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Thanks, @FabienPapet - I agree, I've double-checked, and there's no visible impact.

**Pull request merged**